dref/serializers.py

```diff
@@ -219,3 +274,206 @@ def create(self, validated_data):
+    NestedUpdateMixin,
+    NestedCreateMixin,
+    serializers.ModelSerializer
+):
+    national_society_actions = NationalSocietyActionSerializer(many=True, required=False)
+    needs_identified = IdentifiedNeedSerializer(many=True, required=False)
+    planned_interventions = PlannedInterventionSerializer(many=True, required=False)
+    type_of_onset_display = serializers.CharField(source='get_type_of_onset_display', read_only=True)
+    disaster_category_display = serializers.CharField(source='get_disaster_category_display', read_only=True)
+    created_by_details = UserNameSerializer(source='created_by', read_only=True)
+    images_details = DrefFileSerializer(source='images', many=True, read_only=True)
+    modified_by_details = UserNameSerializer(source='modified_by', read_only=True)
+    disaster_type_details = DisasterTypeSerializer(source='disaster_type', read_only=True)
+    country_district = DrefOperationalUpdateCountryDistrictSerializer(
+        source='drefoperationalupdatecountrydistrict_set',
+        many=True,
+        required=False
+    )
+
+    class Meta:
+        model = DrefOperationalUpdate
+        fields = '__all__'
+        read_only_fields = ('operational_update_number', )
+
+    def validate(self, data):
+        dref = data.get('dref')
+        if not self.instance and dref:
+            dref = get_object_or_404(Dref, id=dref.id)
+            if not dref.is_published:
+                raise serializers.ValidationError(
+                    ugettext('Can\'t create Operational Update for not published %s dref.' % dref.id)
+                )
+            # get the latest dref_operation_update and check whether it is published or not, exclude no operational object created so far
+            operational_update_queryset = DrefOperationalUpdate.objects.filter(dref=dref)
+            if not operational_update_queryset.count() == 0:
+                dref_operational_update = operational_update_queryset.order_by('-operational_update_number')[0]
+                if dref_operational_update and not dref_operational_update.is_published:
+                    raise serializers.ValidationError(
+                        ugettext(
+                            'Can\'t create Operational Update for not published Operational Update %s id and Operational Update Number %i.'
+                            % (dref_operational_update.id, dref_operational_update.operational_update_number)
+                        )
+                    )
+        return data
+
+    def get_total_timeframe(self, start_date, end_date):
+        if start_date and end_date:
+            start_date_month = datetime.datetime.strftime('%m')
+            end_date_month = datetime.datetime.strptime('%m')
+            return abs(end_date_month - start_date_month)
+        return None
+
+    def create(self, validated_data):
+        dref = validated_data.get('dref')
+        dref_operational_update = DrefOperationalUpdate.objects.filter(dref=dref)
```

Comment:
Since we are using `dref_operational_update` here, let's fetch operational_object here instead of count and fetch.